Adds the recommended scenarios for different needs cases. (#180)
mongodb/docs-atlas-architecture: source/deployment-paradigms.txt

Availability Considerations
---------------------------
{+service+} has replication built into your deployments, which means: 

- Database instances are kept tightly in sync with each other, typically in the 
  range of milliseconds.
- In the event of an outage, failover between database instances is fully automatic. 
  It requires no human intervention, and takes only seconds.
- There is no data loss during a failover, including operations that are in 
  flight while the failover happens. If 
  :manual:`writeConcern </reference/write-concern>` is set to ``majority``,  
  these operations are retried after the failover is complete, to ensure that no 
  writes are lost.

Availability is often considered in terms of how fast your application can recover 
from an outage (RTO) and how much data might be lost in an outage (RPO). Because 
all {+service+} instances always have current data, failovers do not 
require restoring of backups.

.. note::

   The maximum RTO and RPO should be considered holistically across the entire 
   application, including component pieces, and not just a single database 
   component. 

Deployment Paradigm Comparison
------------------------------

> To figure out which deployment pattern is right for you, it is necessary to 

Reviewer: ```suggestion
To figure out which deployment pattern is right for you, 
```

break down your applications by how critical they are to your core business. The 

> more important the application (in other words, the more money lost due to an 
> outage), the more you will want to spend on an architecture to automatically 
> handle any outage event.

Reviewer: ```suggestion
more important the application (in other words, the more impact an 
outage has on your business), the more you should consider an architecture that automatically 
handles any outage event.
```

The following table provides a comparison of deployment paradigms to help you 
determine the best fit for your needs: 

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 20 20 

   * - Priority Level
     - Description
     - RTO
     - Deployment Model
     - Relative Cost

   * - **Tier 1**
     - Highest criticality apps. Requires fully automated failover even in the 
       event of regional outages.
     - 0
     - :ref:`3 Regions with 5 or more nodes <arch-center-multi-region-5+3>`
     - $$$

   * - **Tier 2**

>      - Lower criticality apps. Can experience some downtime or maintenance windows 

Reviewer: ```suggestion
     - Lower criticality applications. Can experience some downtime or maintenance windows 
```

       without significant revenue impact.

>      - > 1 hr. and < 8 hr.

Reviewer: ```suggestion
     - > 1 hour and < 8 hours
```

     - | :ref:`2 regions with 3 nodes and backups <arch-center-multi-region-5+2>` or
       | :ref:`3 regions with 1 node in each <arch-center-multi-region-3+3>`
     - $$

   * - **Tier 3**

>      - Lowest criticality apps. Can be done for 24 hrs without significant 

Reviewer: ```suggestion
     - Lowest criticality applications. Can be down for 24 hours without significant 
```

       revenue impact.

>      - > 8 hr.

Reviewer: ```suggestion
     - > 8 hours
```

     - :ref:`3 nodes in a single region <arch-center-paradigms-single>`
     - $+

   * - **Non-Production**

>      - Not critical. Environments that are not directly responsible for 

Reviewer: ```suggestion
     - Noncritical applications. Environments that are not directly responsible for 
```

>        revenue and are not customer-facing. Typically Dev and Test environments.

Reviewer: ```suggestion
       revenue and are not customer-facing. Typically development and testing environments.
```
